lib/membrane/rtp/rtx_parser.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Membrane.RTP.RtxParser do
+  @moduledoc """
+  An element responsible for handling retransmission packets (`rtx`) defined in
+  [RFC 4588](https://datatracker.ietf.org/doc/html/rfc4588#section-4).
+
+  It parses rtx packet and recreates the lost packet
+  """
+
+  use Membrane.Filter
+
+  require Membrane.Logger
+
+  alias Membrane.{Buffer, RTP}
+
+  def_input_pad :input, caps: RTP, demand_mode: :auto
+  def_output_pad :output, caps: RTP, demand_mode: :auto
+
+  def_options rtx_payload_type: [
+                description: "Payload type of retransmission RTP stream"
+              ],
+              payload_type: [
+                description:
+                  "Payload type of original RTP stream that is retransmitted via this SSRC"
+              ]
+
+  @impl true
+  def handle_init(opts) do
+    {:ok, Map.from_struct(opts)}
+  end
+
+  @impl true
+  def handle_caps(:input, rtp_caps, _ctx, state) do
+    {{:ok, forward: rtp_caps}, state}
+  end
+
+  @impl true
+  def handle_process(:input, %Buffer{payload: ""}, _ctx, state) do
+    # Ignore empty buffers, most likely used for bandwidth estimation
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_process(:input, %Buffer{} = buffer, _ctx, state) do
+    <<original_seq_num::16, original_payload::binary>> = buffer.payload
+
+    Membrane.Logger.debug(
+      "[RTX SSRC: #{buffer.metadata.rtp.ssrc}] got retransmitted packet with seq_num #{original_seq_num}"
+    )
+
+    recreated_buffer = %Buffer{
+      buffer
+      | payload: original_payload,
+        metadata: %{
+          rtp: %{
+            buffer.metadata.rtp
+            | sequence_number: original_seq_num,
+              payload_type: state.payload_type
+          }
+        }
```

Comment:
Most of the fields should stay the same, i.e.`csrcs, marker, padding_size, timestamp, total_header_size`. I wondered about the SSRC but it seemed potentially useful to keep the origin of the packet in that form as it will be rewritten while sending anyway, but I wasn't sure about it.

Extensions weren't problematic until enabling simulcast and you're right, I need to do something about it - while `mid` can stay, RTX uses repaired-rid instead of rid and that means one more thing has to be passed from webrtc plugin :disappointed: